### DISCUSSION TITLE
Is it possible to disable console errors when being redirected away from the app?

### DISCUSSION THREAD

**Discussion by erichartline** [vercel/next.js#51196] [Help]
### Summary

I have a bunch of redirects set up in my `next.config.js` file. When I click a link in my app that matches the URL set in my redirect, I get the following error in my console:

`Error: Failed to lookup route: /foobar`

These are filling up my external telemetry when a bunch of users are clicking the same links. Is it possible to suppress these console errors?

**Comment by devjiwonchoi** [comment]
If the route paths do not change often, try to add `permanent: true` option in order to cache the redirect route.

> if true will use the 308 status code which instructs clients/search engines to cache the redirect forever

```js
module.exports = {
  async redirects() {
    return [
      {
        source: '/about',
        destination: '/',
        permanent: true, // ---> here
      },
    ]
  },
}
```

Note that the `redirects` require other features such as Middleware in order for client-side routing like `Link`.

> Redirects are not applied to client-side routing (Link, router.push), unless [Middleware](https://nextjs.org/docs/app/building-your-application/routing/middleware) is present and matches the path.

## Reference

- [Redirects](https://nextjs.org/docs/app/api-reference/next-config-js/redirects)

**Reply by erichartline** [reply]
I'll explore the middleware option to see if that will solve it. What's interesting is that these redirects are working perfectly fine for client-side routing, it's just that they are producing these console errors every single time.
